cmd/limactl/volume.go

```diff
@@ -0,0 +1,235 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"text/tabwriter"
+
+	"github.com/docker/go-units"
+	"github.com/lima-vm/lima/pkg/qemu"
+	"github.com/lima-vm/lima/pkg/store"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func newVolumeCommand() *cobra.Command {
+	var volumeCommand = &cobra.Command{
+		Use:   "volume",
+		Short: "Lima volume management",
+		Example: `  Create a volume:
+  $ limactl volume create data_volume
+
+  List existing volumes:
+  $ limactl volume ls
+
+  Delete a volume:
+  $ limactl volume delete data_volume`,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+	volumeCommand.AddCommand(
+		newVolumeCreateCommand(),
+		newVolumeListCommand(),
+		newVolumeDeleteCommand(),
+		newVolumeUnlockCommand(),
+	)
+	return volumeCommand
+}
+
+func newVolumeCreateCommand() *cobra.Command {
+	var volumeCreateCommand = &cobra.Command{
+		Use: "create VOLUME",
+		Example: `
+To create a new volume:
+$ limactl volume create VOLUME
+`,
+		Short: "Create a Lima volume",
+		Args:  cobra.ExactArgs(1),
+		RunE:  volumeCreateAction,
+	}
+	volumeCreateCommand.Flags().String("size", "2G", "configure the volume size")
+	return volumeCreateCommand
+}
+
+func volumeCreateAction(cmd *cobra.Command, args []string) error {
+	_, err := createVolume(cmd, args)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createVolume(cmd *cobra.Command, args []string) (*store.Volume, error) {
+	// only exactly one arg is allowed (?)
+	name := args[0]
+
+	volDir, err := store.VolumeDir(name)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := os.Stat(volDir); !errors.Is(err, fs.ErrNotExist) {
+		return nil, fmt.Errorf("volume %q already exists (%q)", name, volDir)
+	}
+
+	logrus.Infof("Creating a volume %q", name)
+
+	if err := os.MkdirAll(volDir, 0700); err != nil {
+		return nil, err
+	}
+
+	size, err := cmd.Flags().GetString("size")
+	if err != nil {
+		return nil, err
+	}
+	diskSize, err := units.RAMInBytes(size)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := qemu.CreateDataDisk(volDir, int(diskSize)); err != nil {
+		return nil, err
+	}
+
+	vol, err := store.InspectVolume(name)
+	if err != nil {
+		return nil, err
+	}
+	return vol, nil
+}
+
+func newVolumeListCommand() *cobra.Command {
+	var volumeListCommand = &cobra.Command{
+		Use: "list",
+		Example: `
+To list existing volumes:
+$ limactl volume list
+`,
+		Short:   "List existing Lima volumes",
+		Aliases: []string{"ls"},
+		Args:    cobra.NoArgs,
+		RunE:    volumeListAction,
+	}
+	return volumeListCommand
```

Comment:
This can of course be added later, but it would be nice to have a `--json` flag, just like on the main `list` command:

```console
$ l ls --json | jq
{
  "name": "alpine",
  "status": "Stopped",
  "dir": "/Users/jan/.lima/alpine",
  "arch": "x86_64",
  "cpuType": "host",
  "cpus": 4,
  "memory": 4294967296,
  "disk": 107374182400
}
```